packages/sw-runtime-caching/src/lib/request-wrapper.js

```diff
@@ -117,7 +120,16 @@ class RequestWrapper {
     assert.atLeastOne({request});
 
     const cache = await this.getCache();
-    return await cache.match(request, this.matchOptions);
+    const response = await cache.match(request, this.matchOptions);
+
+    // Give the cacheWillMatch callbacks a chance to modify the cached response.
+    // TODO: Right now, we run all the callbacks sequentially, and return the
+    // final result. Instead, it might make sense to enforce a limit of a single
+    // cacheWillMatch callback, and return an error if there are multiple ones.
+    const callbacks = this.behaviorCallbacks.cacheWillMatch || [];
+    return callbacks.reduce((cachedResponse, callback) => {
+      return callback({cachedResponse});
```

Comment:
This feels like a really weird loop and I think it's the naming.

An array of 'cacheWillMatch' functions, sounds like an array of functions that will return true or false if the cache will match, but it's not it's a function that filters the cache response.

Then looping over some callbacks will get the actual response, others will get null from the previous cached response.

If the previousResponse is null, there is no point calling the callback, it'll have no information to look at. If you agree with this, then this should be a for loop that loops over until it gets a null and then just return early.